src/wert-module.tsx

```diff
@@ -1,72 +1,37 @@
-import * as React from 'react';
-import WertWidget from '@wert-io/widget-initializer';
-
-declare type WertPassThroughProps = React.ComponentProps<'div'>;
-declare type WertModuleProps = WertPassThroughProps & {
-  options: WertWidget["options"]
-};
-declare type WertModuleState = {
-  passThroughProps: WertPassThroughProps
-};
-
-class WertModule extends React.Component<WertModuleProps, WertModuleState> {
-  wertWidget: WertWidget;
-
-  constructor(props: WertModuleProps) {
-    super(props);
-
-    const defaultContainerId = 'wert-module';
-    const wertWidget = new WertWidget({
-      container_id: defaultContainerId,
-      ...props.options,
+import { useRef } from "react";
+import WertWidget from "@wert-io/widget-initializer";
+import generateSignedData from "./generate-signed-data";
+import type { SmartContractOptions } from "./generate-signed-data";
+
+type WidgetOptions = ConstructorParameters<typeof WertWidget>[0];
+type ThemeOptions = Parameters<WertWidget["setTheme"]>[0];
+
+export function useWertWidget(
+  options: WidgetOptions,
+  smartContractOptions?: SmartContractOptions
+) {
+  const wertWidget = useRef<WertWidget | null>(null);
+
+  const initWidget = () => {
+    if (wertWidget.current) wertWidget.current.destroy();
+
+    wertWidget.current = new WertWidget({
+      ...options,
+      ...(smartContractOptions ? generateSignedData(smartContractOptions) : {}),
     });
-    const passThroughProps = {
-      id: props.options.container_id || defaultContainerId,
-      ...props,
-      options: undefined,
-    } as WertPassThroughProps;
-
-    this.wertWidget = wertWidget;
-    this.state = {
-      passThroughProps,
-    };
-  }
-
-  componentDidMount() {
-    this.wertWidget.mount();
-  }
-
-  componentDidUpdate(prevProps: WertModuleProps) {
-    const themeChanged = this.props.options.theme !== prevProps.options.theme;
-    const newColors = Object.keys(this.props.options)
-      .filter(key => key.startsWith('color'))
-      .reduce((accum, key) => {
-        const newColor = this.props.options[key];
-        const oldColor = prevProps.options[key];
-
-        if (newColor !== oldColor) {
-          accum[key] = newColor;
-        }
-
-        return accum;
-      }, {});
-    const colorsChanged = !!Object.keys(newColors).length;
-
-    if (themeChanged || colorsChanged) this.wertWidget.setTheme({
-      theme: this.props.options.theme,
-      colors: newColors,
-    });
-  }
-
-  componentWillUnmount() {
-    this.wertWidget.destroy();
-  }
-
-  render() {
-    return (
-      <div {...this.state.passThroughProps} />
-    );
-  }
+  };
+
+  return {
+    mountWertWidget: () => {
+      initWidget();
+      wertWidget.current?.mount();
+    },
+    destroyWertWidget: () => wertWidget.current?.destroy?.(),
+    getEmbedCode: () => wertWidget.current?.getEmbedCode(),
```

Comment:
This method [is not private ](https://github.com/wert-io/widget-initializer/blob/master/index.ts#L190)in the widget initializer, so I added it here too, however, it is not mentioned in the main [readme](https://github.com/wert-io/widget-initializer#configuration-object-methods). Should it be accessible?